aten/src/ATen/core/jit_type.h

```diff
@@ -622,8 +645,10 @@ struct CAFFE2_API TupleType : public Type {
     }
     return true;
   }
+
   std::vector<TypePtr> elements_;
   bool has_free_variables_;
+  OptNameList names_;
```

Comment:
Hmmm I think it would be cleaner to have a separate type for named tuples

Reply:
@apaszke Since TupleType with names has been merged, I can refactor that part in later PR, this PR would focus on allowing namedtuple to be input of a scripted function.